tests/model/test_model.py

```diff
@@ -561,6 +561,62 @@ def test_topics_in_stream(self, mocker, model, topics_index, fetched, stream_id=
         assert model.index["topics"][stream_id] == return_value
         assert model.index["topics"][stream_id] is not return_value
 
+    @pytest.mark.parametrize(
+        "response, expected_return_value",
+        [
+            (
+                {"result": "success", "msg": "", "email": "username@example.com"},
+                "username@example.com",
+            ),
+            (
+                {"result": "error", "msg": "Invalid stream ID", "code": "BAD_REQUEST"},
+                None,
+            ),
+        ],
+    )
+    def test__fetch_stream_email_address(
+        self, mocker, response, model, expected_return_value, stream_id=1
+    ) -> None:
+        self.client.call_endpoint = mocker.Mock(return_value=response)
+
+        result = model._fetch_stream_email_address(stream_id)
+
+        self.client.call_endpoint.assert_called_once_with(
+            f"/streams/{stream_id}/email_address", method="GET"
+        )
+        assert result == expected_return_value
+
+    @pytest.mark.parametrize(
```

Comment:
Would using the stream_dict fixture help simplify the parametrize for this, or complicate it?

Reply:
Since this test only concerns the presence of the stream email_address, using the Fixture might complicate things (and be a bit unnecessary). However, I'll wait for @neiljp's opinion on this.

Reply:
I think this would likely complicate it, since we wish to have the option to completely remove the email address from the dict, based on the parametrize.

This may become more complex when typing the tests, as the small dict fragment we're using here is not anything like a full Stream or Subscription object.